js/packages/components/chat/MultiMembersSettingsAddMembers.tsx

```diff
@@ -0,0 +1,172 @@
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { FooterCreateGroup } from '@berty-tech/components/main/CreateGroupFooter'
+import { Header, MemberList, AddMembers } from '@berty-tech/components/main/CreateGroupAddMembers'
+import {
+	useContactList,
+	useConversation,
+	useConvMemberList,
+	useMsgrContext,
+} from '@berty-tech/store/hooks'
+import beapi from '@berty-tech/api'
+
+const _iconArrowBackSize = 30
+const _titleSize = 25
+
+export const AddMembersHeader: React.FC<{}> = () => {
+	const navigation = useNavigation()
+	const [{ color, padding, margin, text }, { scaleSize }] = useStyles()
+	const { t }: { t: any } = useTranslation()
+	return (
+		<View
+			style={[
+				padding.medium,
+				margin.bottom.small,
+				{
+					flexDirection: 'row',
+					justifyContent: 'space-between',
+					alignItems: 'center',
+				},
+			]}
+		>
+			<View
+				style={[
+					{
+						flexDirection: 'row',
+						alignItems: 'center',
+					},
+				]}
+			>
+				<TouchableOpacity
+					onPress={navigation.goBack}
+					style={{
+						padding: 7,
+						alignItems: 'center',
+						justifyContent: 'center',
+					}}
+				>
+					<Icon
+						name='arrow-back-outline'
+						width={_iconArrowBackSize * scaleSize}
+						height={_iconArrowBackSize * scaleSize}
+						fill={color.white}
+					/>
+				</TouchableOpacity>
+				<Text
+					style={[
+						text.size.huge,
+						{
+							fontWeight: '700',
+							lineHeight: 1.25 * _titleSize,
+							marginLeft: 10,
+							color: color.white,
+						},
+					]}
+				>
+					{t('chat.add-members.members')}
+				</Text>
+			</View>
+			<Icon
+				name='users'
+				pack='custom'
+				width={35 * scaleSize}
+				height={35 * scaleSize}
+				fill={color.white}
+			/>
+		</View>
+	)
+}
+
+export const MultiMemberSettingsAddMembers: React.FC<{
+	route: { params: { convPK: string } }
+}> = ({ route }) => {
+	const [{ flex, background, margin, color }, { scaleHeight }] = useStyles()
+	const navigation = useNavigation()
+	const ctx = useMsgrContext()
+	const { t }: { t: any } = useTranslation()
+
+	const conv = useConversation(route.params.convPK)
+	const convMembers = useConvMemberList(route.params.convPK)
+	const initialMembers = convMembers.filter((member) => !member.isMe)
+	const [members, setMembers] = useState(initialMembers)
+	const accountContacts = useContactList()
+
+	const setMember = (contact: any) => {
+		if (members.find((member) => member.publicKey === contact.publicKey)) {
+			return
+		}
+		setMembers([...members, contact])
+	}
+	const removeMember = (id: string) => {
+		const filtered = members.filter((member) => member.publicKey !== id)
+		if (filtered.length !== members.length) {
+			setMembers(filtered)
+		}
+	}
+
+	const buf = beapi.messenger.AppMessage.GroupInvitation.encode({ link: conv?.link }).finish()
+	const invitationsToGroup = React.useCallback(async () => {
+		try {
+			await members.forEach((member) => {
```

Comment:
```suggestion
	const invitationsToGroup = React.useCallback(async () => {
		try {
		  const buf = beapi.messenger.AppMessage.GroupInvitation.encode({ link: conv?.link }).finish()
			await members.forEach((member) => {
```
otherwise the callback will always be recomputed

Reply:
you could also `useMemo`